packages/server/lib/automation/commands/key_press.ts

```diff
@@ -91,16 +74,42 @@ async function getActiveWindow (client: Client) {
   }
 }
 
-export async function bidiKeyPress ({ key }: KeyPressParams, client: Client, autContext: string, idSuffix?: string): Promise<void> {
-  const value = BIDI_VALUE[key]
-
-  if (!value) {
-    throw new InvalidKeyError(key)
-  }
+// While other browsers support the named keys, BiDi does not.
+// We need to override the codepoints for the named keys to work.
+export const BidiOverrideCodepoints: Record<SupportedNamedKey, string> = {
+  'ArrowDown': '\uE015',
+  'ArrowLeft': '\uE012',
+  'ArrowRight': '\uE014',
+  'ArrowUp': '\uE013',
+  'End': '\uE010',
+  'Home': '\uE011',
+  'PageDown': '\uE00F',
+  'PageUp': '\uE00E',
+  'Enter': '\uE007',
+  'Tab': '\uE004',
+  'Backspace': '\uE003',
+  'Delete': '\uE017',
+  'Insert': '\uE016',
+  'F1': '\uE031',
+  'F2': '\uE032',
+  'F3': '\uE033',
+  'F4': '\uE034',
+  'F5': '\uE035',
+  'F6': '\uE036',
+  'F7': '\uE037',
+  'F8': '\uE038',
+  'F9': '\uE039',
+  'F10': '\uE03A',
+  'F11': '\uE03B',
+  'F12': '\uE03C',
+}
 
+export async function bidiKeyPress (inKey: SupportedKey, client: Client, autContext: string, idSuffix?: string): Promise<void> {
   const activeWindow = await getActiveWindow(client)
   const { contexts: [{ context: topLevelContext }] } = await client.browsingContextGetTree({})
 
+  const key: string = BidiOverrideCodepoints[inKey] ?? inKey
```

Comment:
I was going to suggest removing the `string` type, since it should be inferred. However, I noticed that the type of `inKey` is being resolved as `never`. Could that be an issue?

I also see that variables typed as `SupportedKey` in `packages/server/lib/automation/commands/key_press.ts` are being resolved to `never` as well. I’m wondering if any type errors would appear if these variables were not resolving to `never`.